src/test/mochitest/browser_dbg-pause-exceptions.js

```diff
@@ -19,26 +19,34 @@ function caughtException() {
 add_task(async function() {
   const dbg = await initDebugger("doc-exceptions.html");
 
-  // test skipping an uncaught exception
+  log("1. test skipping an uncaught exception");
   await uncaughtException();
   ok(!isPaused(dbg));
 
-  // Test pausing on an uncaught exception
+  log("2. Test pausing on an uncaught exception");
   await togglePauseOnExceptions(dbg, true, false);
   uncaughtException();
   await waitForPaused(dbg);
   assertPausedLocation(dbg);
   await resume(dbg);
+  await waitForActive(dbg);
 
-  // Test pausing on a caught Error
+  log("3. Test pausing on a caught Error");
   caughtException();
   await waitForPaused(dbg);
   assertPausedLocation(dbg);
+
+  log("3.b Test pausing in the catch statement");
+  await resume(dbg);
+  await waitForPaused(dbg);
+  assertPausedLocation(dbg);
```

Comment:
we had previously been sloppy w/ this test and had allowed `togglePauseOnExceptions` to do an implicit resume, which was creating a race condition.